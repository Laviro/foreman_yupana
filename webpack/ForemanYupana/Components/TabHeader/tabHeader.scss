.tab-header {
  h1 {
    margin-top: 0;
  }

  button {
    float: right;
  }
}
