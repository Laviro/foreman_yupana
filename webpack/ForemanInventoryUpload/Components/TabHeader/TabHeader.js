import React from 'react';
import PropTypes from 'prop-types';
import { Grid, Button, Icon, DropdownItem, Dropdown, DropdownToggle, DropdownToggleAction } from 'patternfly-react';
import { noop } from 'foremanReact/common/helpers';
import { sprintf, translate as __ } from 'foremanReact/common/I18n';
import { isExitCodeLoading } from '../../ForemanInventoryHelpers';
import './tabHeader.scss';

const onActionToggle = () => {
  setIsActionOpen(prev => !prev);
};

const dropdownItems = [
  <DropdownItem
    aria-label="restart_disconnected"
    ouiaId="restart_disconnected"
    key="restart_disconnected"
    component="button"
    onClick={restartDisconnected}
    isDisabled={isExitCodeLoading(exitCode)}
  >
    {__('Restart without uploading')}
  </DropdownItem>,
];

const TabHeader = ({ exitCode, onRestart, onDownload, restartDisconnected, toggleFullScreen }) => (
  <Grid.Row className="tab-header">
    <Grid.Col sm={6}>
      <p>{sprintf(__('Exit Code: %s'), exitCode)}</p>
    </Grid.Col>
    <Grid.Col sm={6}>
      <div className="tab-action-buttons">
        {onRestart ? (
          <Dropdown
          aria-label="restart_dropdown"
          ouiaId="restart_dropdown"
          toggle={
            <DropdownToggle
              aria-label="restart_report_toggle"
              ouiaId="restart_report_toggle"
              splitButtonItems={[
                <DropdownToggleAction key="action" aria-label="bulk_actions" onClick={onRestart}>
                  {__('Restart')}
                </DropdownToggleAction>,
              ]}
              splitButtonVariant="action"
              toggleVariant="primary"
              onToggle={onActionToggle}
              isDisabled={isExitCodeLoading(exitCode)}
            />
        }
          isOpen={isActionOpen}
          dropdownItems={dropdownItems}
        />
        ) : null}
        {onDownload ? (
          <Button onClick={onDownload}>
            {__('Download Report')} <Icon name="download" />
          </Button>
        ) : null}
        <Button onClick={toggleFullScreen}>
          {__('Full Screen')}
          <Icon name="arrows-alt" />
        </Button>
      </div>
    </Grid.Col>
  </Grid.Row>
);

TabHeader.propTypes = {
  onRestart: PropTypes.func,
  onDownload: PropTypes.func,
  exitCode: PropTypes.string,
  toggleFullScreen: PropTypes.func,
};

TabHeader.defaultProps = {
  onRestart: null,
  exitCode: '',
  onDownload: null,
  toggleFullScreen: noop,
};

export default TabHeader;
