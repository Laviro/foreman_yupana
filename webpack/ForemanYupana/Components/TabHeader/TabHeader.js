import React from 'react';
import PropTypes from 'prop-types';
import { noop, Grid, Button } from 'patternfly-react';
import './tabHeader.scss';

const TabHeader = ({ exitCode, onRestart }) => (
  <Grid.Row className="tab-header">
    <Grid.Col sm={8}>
      <h1># Exit Code: {exitCode}</h1>
    </Grid.Col>
    <Grid.Col sm={4}>
      <Button bsStyle="primary" onClick={onRestart}>
        Restart
      </Button>
    </Grid.Col>
  </Grid.Row>
);

TabHeader.propTypes = {
  onRestart: PropTypes.func,
  exitCode: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
};

TabHeader.defaultProps = {
  onRestart: noop,
  exitCode: '',
};

export default TabHeader;
