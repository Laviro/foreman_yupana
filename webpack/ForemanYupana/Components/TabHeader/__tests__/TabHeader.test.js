import { testComponentSnapshotsWithFixtures } from 'react-redux-test-utils';

import TabHeader from '../TabHeader';

const fixtures = {
  'render without Props': {},
  /** fixtures, props for the component */
};

describe('TabHeader', () => {
  describe('rendering', () =>
    testComponentSnapshotsWithFixtures(TabHeader, fixtures));
});
