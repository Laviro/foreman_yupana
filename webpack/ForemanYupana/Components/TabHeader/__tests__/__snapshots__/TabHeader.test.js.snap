// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabHeader rendering render without Props 1`] = `
<Row
  bsClass="row"
  className="tab-header"
  componentClass="div"
>
  <Col
    bsClass="col"
    componentClass="div"
    sm={8}
  >
    <h1>
      # Exit Code: 
    </h1>
  </Col>
  <Col
    bsClass="col"
    componentClass="div"
    sm={4}
  >
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="primary"
      disabled={false}
      onClick={[Function]}
    >
      Restart
    </Button>
  </Col>
</Row>
`;
